src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartUtils.java

```diff
@@ -0,0 +1,143 @@
+        System.out.println("Port is ready");
+        Mqtt5BlockingClient client = getMqtt5BlockingClient(container.getMappedPort());
+        var publishes = client.publishes(MqttGlobalPublishFilter.ALL);
+        client.subscribeWith().topicFilter("test").send();
+        client.publishWith()
+                .topic("test")
+                .payload(message.getBytes(StandardCharsets.UTF_8))
+                .qos(MqttQos.AT_LEAST_ONCE)
+                .send();
+        Mqtt5Publish receivedMessage = publishes.receive();
+        assertTrue(publishes.receive(200, TimeUnit.MILLISECONDS).isEmpty());
+        return receivedMessage;
+
+    }
+
+    @NotNull
+    private Mqtt5BlockingClient getMqtt5BlockingClient(final int mappedPort) throws RuntimeException {
+        Mqtt5BlockingClient client = Mqtt5Client.builder()
+                .automaticReconnectWithDefaultConfig()
+                .serverPort(mappedPort)
+                .buildBlocking();
+        client.connect();
+        assertEquals(client.getState(), MqttClientState.CONNECTED);
+        System.out.println("Mqtt Client Connected");
+        return client;
+    }
+
+    public @NotNull ExecResult deployChart(File valuesPath,File operatorPath) throws IOException, InterruptedException {
+        //helm --kubeconfig /etc/rancher/k3s/k3s.yaml install hivemq /test/charts/hivemq-operator -f /test/src/test/resources/testValues.yaml
+        var outUpdate = container.execInContainer("/bin/helm",
+                "dependency",
+                "update",
+                operatorPath.getAbsolutePath()+"/");
+        assertTrue(outUpdate.getStderr().isEmpty());
+        return container.execInContainer("/bin/helm",
+                "--kubeconfig",
+                "/etc/rancher/k3s/k3s.yaml",
+                "install",
+                "hivemq",
+                operatorPath.getAbsolutePath(),
+                "-f",
+                valuesPath.getAbsolutePath());
+    }
+
+    public @NotNull String exposeService() throws IOException, ApiException, InterruptedException {
+        String kubeConfigYaml = container.getKubeConfigYaml();
+        var client = Config.fromConfig(new StringReader(kubeConfigYaml));
+        var api = new CoreV1Api(client);
+        var nodes = api.listNode(null, null, null, null, null, null, null, null, null, null);
+        assertTrue(nodes.getItems().size() > 0);
+        V1Pod foundPods = waitForHiveMQCluster(api);
+        assertNotNull(foundPods);
+        //When the pod is running the cluster needs time to be ready
+        var podLog = new PodLogs(client);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (var is = podLog.streamNamespacedPodLog(foundPods)) {
+            var scanner = new Scanner(is).useDelimiter("\n");
+            while (scanner.hasNext()) {
+                var s = scanner.next();
+                stringBuilder.append(s);
+                stringBuilder.append("\n");
+                if (s.contains("Started HiveMQ in")) {
```

Comment:
this is the job of the hivemq operator to make sure the node is ready, it should be enough to just wait until the condition `HiveMQCluster.status.state==Running`
(which does pretty much this)